core/src/main/scala/cats/data/FailFastFuture.scala

```diff
@@ -0,0 +1,35 @@
+package cats.data
+
+import cats.Applicative
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success}
+
+class FailFastFuture[A](val value: Future[A]) extends AnyVal
+
+object FailFastFuture {
+
+  def apply[A](value: Future[A]): FailFastFuture[A] = new FailFastFuture(value)
+
+  def catsDataApplicativeForFailFastFuture(implicit ec: ExecutionContext): Applicative[FailFastFuture] =
```

Comment:
shall we also test the Applicative laws on this? Also want to link @tpolecat's comment [here](https://github.com/typelevel/cats/issues/2015#issuecomment-342206568) that it should limit to `CommutativeApply`

Reply:
Just tested, this doesn't seem to pass the `ApplicativeLaw`.

Reply:
Several of them

Reply:
Good catch! I'll try to see if it might be a valid `Apply` instead.

Reply:
FastFuture.applicative.ap consistent with product + map *** FAILED ***
FastFuture.applicative.apply composition *** FAILED ***
FastFuture.applicative.followedBy consistent map2 *** FAILED ***
FastFuture.applicative.forEffect consistent map2 *** FAILED ***
FastFuture.applicative.map2/map2Eval consistency *** FAILED ***
FastFuture.applicative.map2/product-map consistency *** FAILED *** 
FastFuture.applicative.semigroupal associativity *** FAILED ***